src/item.cpp

```diff
@@ -1516,75 +1479,226 @@ bool item::stacks_with( const item &rhs, bool check_components, bool combine_liq
+           lhs.rotten() == rhs.rotten();
+}
+
+bool _stacks_food_status( item const &lhs, item const &rhs )
+{
+    return ( lhs.is_fresh() == rhs.is_fresh() || lhs.is_going_bad() == rhs.is_going_bad() ||
+             lhs.rotten() == rhs.rotten() ) &&
+           lhs.has_flag( flag_MUSHY ) == rhs.has_flag( flag_MUSHY ) &&
+           lhs.has_own_flag( flag_DIRTY ) == rhs.has_own_flag( flag_DIRTY );
+}
+
+bool _stacks_food_traits( item const &lhs, item const &rhs )
+{
+    return ( !lhs.is_food() && !rhs.is_food() ) ||
+           ( lhs.is_food() && rhs.is_food() &&
+             lhs.has_flag( flag_HIDDEN_POISON ) == rhs.has_flag( flag_HIDDEN_POISON ) &&
+             lhs.has_flag( flag_HIDDEN_HALLU ) == rhs.has_flag( flag_HIDDEN_HALLU ) );
+}
+
+bool _stacks_food_perishable( item const &lhs, item const &rhs, bool check_cat )
+{
+    return !check_cat || ( !lhs.is_food() && !rhs.is_food() ) ||
+           ( lhs.is_food() && rhs.is_food() && lhs.goes_bad() == rhs.goes_bad() &&
+             lhs.has_flag( flag_INEDIBLE ) == rhs.has_flag( flag_INEDIBLE ) );
+}
+
+bool _stacks_clothing_size( item const &lhs, item const &rhs )
+{
+    avatar &u = get_avatar();
+    item::sizing const u_sizing = lhs.get_sizing( u );
+    return u_sizing == rhs.get_sizing( u ) &&
+           ( u_sizing == item::sizing::ignore ||
+             ( lhs.has_flag( flag_FIT ) == rhs.has_flag( flag_FIT ) &&
+               lhs.has_flag( flag_VARSIZE ) == rhs.has_flag( flag_VARSIZE ) ) );
+}
+
+bool _stacks_wetness( item const &lhs, item const &rhs )
+{
+    return lhs.wetness == rhs.wetness || ( ( lhs.wetness > 0 ) == ( rhs.wetness > 0 ) ) ||
+           ( lhs.has_flag( flag_WET ) && rhs.has_flag( flag_WET ) );
+}
+
+bool _stacks_cbm_status( item const &lhs, item const &rhs )
+{
+    return ( !lhs.is_bionic() && !rhs.is_bionic() ) ||
+           ( lhs.is_bionic() && rhs.is_bionic() &&
+             lhs.has_flag( flag_NO_PACKED ) == rhs.has_flag( flag_NO_STERILE ) &&
+             lhs.has_flag( flag_NO_STERILE ) == rhs.has_flag( flag_NO_STERILE ) );
+}
+
+bool _stacks_ethereal( item const &lhs, item const &rhs )
+{
+    static const std::string varname( "ethereal" );
+    return ( !lhs.ethereal && !rhs.ethereal ) ||
+           ( lhs. ethereal && rhs.ethereal && lhs.get_var( varname ) == rhs.get_var( varname ) );
+}
+
+bool _stacks_ups( item const &lhs, item const &rhs )
+{
+    return ( !lhs.is_tool() && !rhs.is_tool() ) ||
+           ( lhs.is_tool() && rhs.is_tool() &&
+             lhs.has_flag( flag_USE_UPS ) == rhs.has_flag( flag_USE_UPS ) );
+}
+
+bool _stacks_mods( item const &lhs, item const &rhs )
+{
+    const std::vector<const item *> this_mods = lhs.mods();
+    const std::vector<const item *> that_mods = rhs.mods();
+    if( this_mods.size() == that_mods.size() ) {
+        std::vector<const item *> diff;
+        std::set_symmetric_difference( this_mods.begin(), this_mods.end(), that_mods.begin(),
```

Comment:
Good catch; thanks. I've restored the original code since these vectors aren't even sorted. Quick test

![Screenshot from 2024-01-14 00-22-12](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/e3970e40-120b-4c5a-94be-61de3e5a9113)